lfsapi/client.go

```diff
@@ -75,6 +77,41 @@ func (c *Client) Do(req *http.Request) (*http.Response, error) {
 	return res, c.handleResponse(res)
 }
 
+func (c *Client) setExtraHeaders(req *http.Request) http.Header {
+	var copy http.Header = make(http.Header)
+	for k, vs := range req.Header {
+		copy[k] = vs
+	}
```

Comment:
Why bother making a copy? Can't you just modify the req headers in place?

Also, why are you using the longer `var` syntax to init the var, instead of `copy := make(http.Header)`?